src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -2813,6 +2822,372 @@ class AlterMaterializedViewStatementSegment(BaseSegment):
     )
 
 
+class CreateFileFormatSegment(BaseSegment):
+    """A snowflake `CREATE FILE FORMAT` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html
+    """
+
+    type = "create_file_format_segment"
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        Sequence("FILE", "FORMAT"),
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+        # TYPE = <FILE_FORMAT> is included in below parameter segments.
+        # It is valid syntax to have TYPE = <FILE_FORMAT> after other parameters.
+        # Below parameters are either Delimited/AnyNumberOf.
+        # Snowflake does allow mixed but this is not supported.
+        # @TODO: Update below when an OptionallyDelimited Class is available.
+        OneOf(
+            Ref("CsvFileFormatTypeParameters"),
+            Ref("JsonFileFormatTypeParameters"),
+            Ref("AvroFileFormatTypeParameters"),
+            Ref("OrcFileFormatTypeParameters"),
+            Ref("ParquetFileFormatTypeParameters"),
+            Ref("XmlFileFormatTypeParameters"),
+        ),
+        Sequence(
+            # Use a Sequence and include an optional CommaSegment here.
+            # This allows a preceding comma when above parameters are delimited.
+            Ref("CommaSegment", optional=True),
```

Comment:
Wouldn't you be better to add `allow_trailing=True` to the `Delimited` sections of each `XXXTypeParameters`?

Reply:
I've left this as it is -

the reason being is that when these segments are used elsewhere (e.g. `ExternalTableSegment`) a trailing comma isn't valid syntax.

AFAIK it is only the `CREATE FILE FORMAT` segment where the trailing comma could be valid - and then, it's only valid when there is a `COMMENT`. I think that makes sense to group it with the `COMMENT` sequence in this case!